zhmccli/_cmd_partition.py

```diff
@@ -916,6 +967,133 @@ def cmd_partition_console(cmd_ctx, cpc_name, partition_name, options):
+
+    if storage_management_feature(partition):
+
+        # Check and parse the --volume option into SG.SV or UUID
+        volume_parts = volume_opt.split('.')
+        if len(volume_parts) == 1:  # format: UUID
+            uuid = volume_parts[0]
+            sg_list = partition.list_attached_storage_groups()
+            storage_volume = None
+            for sg in sg_list:
+                try:
+                    storage_volume = sg.storage_volumes.find(uuid=uuid)
+                    break
+                except zhmcclient.NotFound:
+                    continue
+            if not storage_volume:
+                raise_click_exception(
+                    "Storage volume with UUID '{uuid}' specified in the "
+                    "--volume option was not found in any storage group "
+                    "attached to partition '{part}' of CPC '{cpc}'.".
+                    format(uuid=uuid, part=partition_name, cpc=cpc_name),
+                    cmd_ctx.error_format)
+        elif len(volume_parts) == 2:  # format: SG.SV
+            sg_name, sv_name = volume_parts
+            sg_list = partition.list_attached_storage_groups()
+            sg = None
+            for _sg in sg_list:
+                if _sg.name == sg_name:
+                    sg = _sg
+                    break
+            if not sg:
+                raise_click_exception(
+                    "Storage group '{sg}' specified in the --volume option was "
+                    "not found in the storage groups attached to partition "
+                    "'{part}' of CPC '{cpc}'.".
+                    format(sg=sg_name, part=partition_name, cpc=cpc_name),
+                    cmd_ctx.error_format)
+            try:
+                storage_volume = sg.storage_volumes.find(name=sv_name)
+            except zhmcclient.NotFound:
+                raise_click_exception(
+                    "Storage volume '{sv}' specified in the --volume option "
+                    "was not found in the volumes of storage group '{sg}' "
+                    "attached to partition '{part}' of CPC '{cpc}'.".
+                    format(sv=sv_name, sg=sg_name, part=partition_name,
+                           cpc=cpc_name),
+                    cmd_ctx.error_format)
+        else:
+            raise_click_exception(
+                "Invalid format for --volume option: '{v}'. CPC '{cpc}' has "
+                "the storage management feature which requires the formats "
+                "'SG.SV' or 'UUID'.".
+                format(v=volume_opt, cpc=cpc_name),
+                cmd_ctx.error_format)
+
+        # Set the remaining feature-specific parameters
+        properties['storage-volume-uri'] = storage_volume.uri
+        properties['timeout'] = timeout
+
+        op_properties = {
+            'dump-program-info': properties,
+            'dump-program-type': 'storage',
+        }
+        try:
+            partition.start_dump_program(
```

Comment:
Is it intended that no timeout is specified?

Reply:
I suppose you talk about the `operation_timeout` parameter that is for waiting for the asynchronous job to complete, and not about the `timeout` parameter for a FICON target volume.

To be honest, I did not think about this, but we should have a timeout capability for that.

The question is, can we combine these two timeout values to keep it simple for the users, i.e. have the `--timeout` option determine both timeout values? For now, I have changed the code to do that, but that is probably worth a discussion.

The other question is, do we need to handle the operation timing out (exception `zhmcclient.OperationTimeout`) other than currently where we end the zhmc program with an according error message. This means the asynchronous job will continue to execute, but zhmc provides no means to manage asynchronous jobs. It seems to me that we should keep not interfering with the asynchronously running job, but add a job mgmt capability to zhmc so that such jobs can be determined and managed.

**DISCUSSION**